Hide media card hover buttons on touch devices

andrebk opened the pull request:
## Description of changes

When hovering over the image on media cards, three buttons are shown for tracking, lists, and history. This works great on desktop, but on mobile there is no hover state so they aren't shown. Worse, when tapping the image where the invisible buttons are, you will press the button instead of navigating to the media details page as intended.

This PR hides the entire overlay and the buttons on touch devices (`pointer-coarse`), and adds a link to the `img` itself in addition to the overlay so you can still navigate to the details page when the overlay is hidden. This removes the accidental button clicks on mobile.

andrebk committed:
Hide media card hover buttons on touch devices

FuzzyGrim commented:
> This works great on desktop, but on mobile there is no hover state so they aren't shown. Worse, when tapping the image where the invisible buttons are, you will press the button instead of navigating to the media details page as intended.

This seems like a bug, it should look like this:

https://github.com/user-attachments/assets/35843a6e-c365-4489-a63a-eab4ba1d5db5

This is on iOS, first tap shows the overlay and then you can choose to tap the buttons or the background image to navigate to the details page.

andrebk committed:
Fix integration tests

The integration tests failed because an element selector found two links
instead of one, after adding a link element wrapping the media card
image.

This happened because the element selector was looking for a link with
an "accessible name"  matching the media title. Previously this was only
the media card title, which was a link with the title in it, so
the accessible name matched. But with the new link around the image,
that link also got the title of the media as an accessible name because
the title was used as an alt for the image.

This has been resolved by getting the element by its title instead of its
role, since the title attribute is explicit instead of implicit and is
only set on the desired title link element.

andrebk commented:
> This is on iOS, first tap shows the overlay and then you can choose to tap the buttons or the background image to navigate to the details page.

On android using firefox, that first tap causes navigation to the details page, so there is no time to press any of the buttons. If you then use the back button to return to the page, the overlay and buttons are showing, which is weird, but more in the way than useful.

https://github.com/user-attachments/assets/7fcb1b1a-8426-4552-9f5f-e08b84a13eb5



On iOS, do you need two presses to click the button, or can you still click it directly while invisible without activating the overlay first?

Even if what I'm seeing is not intended behaviour, I think I would still prefer no overlay on mobile. I want the big tappable target (the image) to navigate me, so I don't have to tap the tiny title. And the overlay isn't as much of a shortcut even with your behaviour. It takes two "clicks" to track something both with the overlay (tap to show, tap to open tracking menu) and with the details page (tap to navigate to details page, tap to open tracking menu). On desktop it's more of a shortcut since hovering is "free".

FuzzyGrim commented:
Yes, it still requires two clicks, but the big advantage is that you can stay on the same page without having to go to the details page to edit.

But I can see how some people might prefer your method, I'll merge this and then focus on adding a UI preferences setting to allow both options.
